src/vector/backends/numpy.py

```diff
@@ -20,6 +20,7 @@
 import typing
 
 import numpy
+from numpy._globals import _NoValue, _NoValueType
```

Comment:
Set `keepdims=False` by default.

```suggestion
```

Anything that starts with an underscore is not safe for other libraries to use, so that would include this one.

NumPy's default value for `keepdims` is effectively `False`; that's what's used if no `keepdims` argument is given.

```python
>>> np.sum(np.ones((3, 5)), axis=1)
array([5., 5., 5.])
>>> np.sum(np.ones((3, 5)), axis=1, keepdims=False)
array([5., 5., 5.])
>>> np.sum(np.ones((3, 5)), axis=1, keepdims=True)
array([[5.],
       [5.],
       [5.]])
```

They have this `_NoValue` to distinguish "user did not specify a `keepdims` value" from "user explicitly specified the value that would be the default, anyway." The reason they're making that distinction is given in the [np.sum](https://numpy.org/doc/stable/reference/generated/numpy.sum.html) documentation:

> If the default value is passed, then _keepdims_ will not be passed through to the `sum` method of sub-classes of `ndarray`, however any non-default value will be. If the sub-class’ method does not implement _keepdims_ any exceptions will be raised.

presumably to allow for subclasses that did not handle this argument, maybe because they predated it?

If Vector just sets the default `keepdims=False`, then it would not carry over this behavior to subclasses of `vector.backends.numpy.VectorNumpy*D`, but that's probably fine. We can just declare that while NumPy doesn't expect subclasses of `np.ndarray` to have a `keepdims` argument on its `sum` method, we do expect subclasses of `vector.backends.numpy.VectorNumpy*D` to have such an argument. An inheritance subtree can make stronger assumptions that parents of that subtree.

If this is also the case for other reducers, then this PR should probably fix them as well, taking `keepdims=False` as the default everywhere.

Note: this is how Awkward Array solves the same problem:

https://github.com/scikit-hep/awkward/blob/684a84deffb51c75ba05ec3ca69ad964c3ff5845/src/awkward/operations/ak_sum.py#L15-L23

It handles every argument that NumPy handles, but it does have stronger constraints on them. (Now that I'm looking at it, we seem to be missing `initial`...)

Reply:
> They have this _NoValue to distinguish "user did not specify a keepdims value" from "user explicitly specified the value that would be the default, anyway." The reason they're making that distinction is given in the [np.sum](https://numpy.org/doc/stable/reference/generated/numpy.sum.html) documentation:

Oh, that makes sense, I missed this.

Thanks for the detailed explanation! This was very helpful!